### DISCUSSION TITLE
2006/07/19/%E5%A6%82%E4%BD%95%E4%BD%BF%E7%94%A8-vi-%E7%9A%84%E9%87%8D%E5%85%A5%E5%B1%9E%E6%80%A7%EF%BC%88reentrant%EF%BC%89

### DISCUSSION THREAD

**Discussion by giscus** [ruanqizhen/blog#95] [Announcements]
# 2006/07/19/%E5%A6%82%E4%BD%95%E4%BD%BF%E7%94%A8-vi-%E7%9A%84%E9%87%8D%E5%85%A5%E5%B1%9E%E6%80%A7%EF%BC%88reentrant%EF%BC%89

在 VI Properties -> Execution 中可以选择 VI 的Reentrant Execution属性（中文译为：可重入执行）。 我们在《LabVIEW 程序的内存优化》一文中讨论过，尽量不要把 VI 设置为重入属性，因为这样就多占用了内存，降低了运行效率。此外，如果不加注意的话，还可能引发多线程不安全的问题。 尽管可重入 VI 在 LabVIEW 中不是必须的，但是在某些情况下使用可重入 VI 可以简化我们的程序。那么在什么情况下可以使用 Reentrant VI 呢？

https://blog.qizhen.xyz/2006/07/19/%E5%A6%82%E4%BD%95%E4%BD%BF%E7%94%A8-vi-%E7%9A%84%E9%87%8D%E5%85%A5%E5%B1%9E%E6%80%A7%EF%BC%88reentrant%EF%BC%89

**Comment by ruanqizhen** [comment]
之前博客下的留言：

lcz_1260
2009年03月23日 00:38 编辑
赞一个啦！又学会一种技巧！

回复
Qizhen
2009年02月3日 20:59 编辑
如果你的VI不是很复杂，可以发信给我，我帮你看一下。

回复
颖
2009年01月22日 08:04 编辑
你好，我现在初学Labview,需要调试一个VI,这个Vi曾经能够正常运行的，可是现在总显示有错误，我又找不出来，你能帮我看看吗？

回复
帆
2008年01月17日 10:36 编辑
好强啊 我刚开始学着用~~

回复
Qizhen
2007年07月9日 09:25 编辑
kimi，你可以把你的VI或者截图发给我 ruanqizhen@gmail.com，我看一下才能知道是什么问题。

回复
Zhang Ping
2007年07月7日 15:46 编辑
你好。刚拜读你写的文章，受益匪浅。但是在第二个例子时，图4，计算子VI，我做了一个一样的子VI，但是发现count输出一直是0，我用Probe在各个点进行了测试，程序是运行的，正确的。但是while语句外面的count控件输出一直为零。我特意将while中的i添加了indicator,在while语句内有输出，但在while语句外加上一个indicator，数字仍然没有变化。请问这是什么原因！！！

回复
Unknown
2006年09月16日 19:48 编辑
辛苦了！谢谢
 

回复
Unknown
2006年08月7日 21:50 编辑
又知道了一样，很高兴，谢谢

回复
清扬
2006年07月22日 16:49 编辑
我也是labview的爱好者啊   以后多多交流呀~

回复
yi
2006年07月21日 21:35 编辑
先收藏，再仔细学。谢谢好文章

回复
onemade
2006年07月21日 01:46 编辑
呵呵　同楼上的

回复
洋
2006年07月20日 20:12 编辑
原来是个LABVIEW高手啊~我前段时间的时间就是用LABVIEW写程序~HOHO~有缘啊~以后有这方面的问题就来问你啦~
